pandas/indexes/base.py

```diff
@@ -86,6 +86,12 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    # required for backward compat, because PI can't be instantiated with
+    # ordinals through __new__ GH #13277
+    if issubclass(cls, ABCPeriodIndex):
+        if d['data'].dtype == 'int64':
+            values = d.pop('data')
```

Comment:
I think its worth putting this in place (but obviously for PeriodIndex) in ``__reduce__``, so that current/future pickling will directly call ``_new_PeriodIndex``, e.g.: https://github.com/pandas-dev/pandas/blob/master/pandas/tseries/index.py#L144

I think you *still* will need this, but you can then simplify to:

```
if issubclass(cls, 'ABCPeriodIndex):
    from pandas.tseries.period import _new_PeriodIndex
   return _new_PeriodIndex(cls, **d)
return cls.__new__(cls, **d)
```

which moves the actual logic to the period.py (where it belongs).

also add a TODO to remove this in a future version (from Index).